test-cypress/integration/repository/repositories.spec.js

```diff
@@ -0,0 +1,404 @@
+        saveRepository();
+
+        getRepositoryCreateForm().should('be.visible');
+        getRepositoryIdField().should('have.class', 'ng-invalid');
+
+        getToast()
+            .find('.toast-error')
+            .should('be.visible')
+            .and('contain', 'Repository id cannot be empty');
+    });
+
+    it('should allow creation of repositories with custom settings', () => {
+        const repoTitle = 'Repo title for ' + repositoryId;
+        const newBaseUrl = 'http://example.org/wine#';
+
+        createRepository();
+
+        getRepositoryIdField().type(repositoryId);
+        getRepositoryTitleField()
+            .should('have.value', '')
+            .type(repoTitle)
+            .should('have.value', repoTitle);
+
+        // Should be disabled before changing the rule set
+        getRepositoryDisableSameAsCheckbox()
+            .should('have.value', 'true'); // true as a string..
+
+        // RDFS-Plus (Optimized) -> OWL-Horst (Optimized)
+        getRepositoryRulesetMenu()
+            .should('have.value', '7')
+            .select('8')
+            .should('have.value', '8');
+
+        // Should be automatically enabled when the rule set is changed to one of the OWL rule set
+        getRepositoryDisableSameAsCheckbox()
+            .should('have.value', 'false');
+
+        getRepositoryBaseURLField()
+            .should('have.value', 'http://example.org/owlim#')
+            .clear()
+            .type(newBaseUrl)
+            .should('have.value', newBaseUrl);
+
+        getRepositoryContextIndexCheckbox()
+            .should('have.value', 'false')
+            .check()
+            .should('have.value', 'true');
+
+        saveRepository();
+
+        // Go to edit and check if everything is properly saved
+        editRepository(repositoryId);
+
+        cy.url().should('include', '/repository/edit/' + repositoryId);
+
+        getRepositoryCreateForm().should('be.visible');
+        getRepositoryIdField().should('have.value', repositoryId);
+        getRepositoryTitleField().should('have.value', repoTitle);
+        // OWL-Horst (Optimized) has become 9
+        getRepositoryRulesetMenu().should('have.value', '9');
+        getRepositoryDisableSameAsCheckbox().should('have.value', 'false');
+        getRepositoryBaseURLField().should('have.value', newBaseUrl);
+        getRepositoryContextIndexCheckbox().should('have.value', 'true');
+    });
+
+    it('should allow to switch between repositories', () => {
+        const secondRepoId = 'second-repo-' + Date.now();
+        createRepository();
+        typeRepositoryId(repositoryId);
+        saveRepository();
+
+        createRepository();
+        typeRepositoryId(secondRepoId);
+        saveRepository();
+
+        // Connect to the first repo via the connection icon
+        // Note: Not using within() because the whole row will be re-rendered & detached
+        getRepositoryFromList(repositoryId)
+            .find('.icon-connection-off')
```

Comment:
This could be extracted in a function eventually or at least make the selector a constant. There are some other similar repetitions for other elements in the tests above also.
```javascript
getRepositoryFromList(repositoryId).find('.icon-connection-off')
```

Reply:
Added two reusable functions